#163501151 Register as candidate

Rexben001 opened the pull request:
### What does this PR do?
Register as a candidate route

### Description of Task to be completed?
- Create a route to get a register a candidate
- Secure route from non-users
- Write tests to test the route
- Refactor the code base

### How should this be manually tested?
 `git clone` this repo
`npm install` to install the packages
`npm run start-dev` to run the app
`npm test` to run the test cases

### Any background context you want to provide
## API endpoint
POST `/api/v1/office/:user_id/register` to register a user

### What are the relevant pivotal tracker stories?
#163501151

Rexben001 committed:
[Feature #163501151] Create candidate table

Rexben001 committed:
[Feature #163501151] Create method to create a contestant

Rexben001 committed:
[Feature #163501151] Validate input fields

Rexben001 committed:
[Feature #163501151] Create routes to register candidate

Rexben001 committed:
[Feature #163501151] Update the response body

Rexben001 committed:
[Feature #163501151] Update the validator

Rexben001 committed:
[Feature #163501151] Fixed the failing test

Rexben001 committed:
[Feature #163501151] Create two database, seed the database

Rexben001 committed:
[Feature #163501151] Add .nyc_output & dist folder

Rexben001 committed:
[Feature #163501151] Add .nyc_output & dist folder

Rexben001 committed:
[Feature #163501151] Add .nyc_output & dist folder

Rexben001 committed:
[Feature #163501151] Removed the dropTable()

Rexben001 committed:
[Feature #163501151] Update start-dev script

Rexben001 committed:
[Feature #163501151] Update app.js

Rexben001 committed:
[Feature #163501151] Rewrite tests

Rexben001 committed:
[Feature #163501151] Update the tables

kenware commented on .gitignore:
```diff
@@ -59,3 +59,7 @@ typings/
+
+.nyc_output/
+
+dist
```
Add a new line here at the end of the file

Rexben001 replied:
Ok
Thanks

Rexben001 committed:
[Feature #163501151] Remove SET

kenware commented on server/models/database.js:
```diff
@@ -86,4 +89,27 @@ const office = async () => {
+    FOREIGN KEY (party) REFERENCES parties(party_id),
+    FOREIGN KEY (createdBy) REFERENCES users(user_id),
+    FOREIGN KEY (office) REFERENCES offices(office_id),
+    PRIMARY KEY (createdBy, office)
```
Will this two primary key ensure that candidate is not created twice?

Rexben001 replied:
Yes, it will

kenware commented on server/controllers/userController.js:
```diff
@@ -112,25 +112,65 @@ class UserControllers {
+   */
+  static contestInElection(req, res) {
+    try {
+      const id = Number(req.params.user_id);
```
I need you to validate that it is only admin that make this API call

Rexben001 replied:
Yes, I was thinking of creating all the endpoints but securing routes.
Is it actually okay with you?

kenware replied:
I did not see where you are securing the route. Normally you should have authentication middleware which will be added to the route

Rexben001 committed:
[Feature #163501151] Add ssl=true

Rexben001 committed:
[Feature #163501151] Add a new line here at the end of the file

Rexben001 committed:
[Feature #163501151] Update .travis to create a db on TravisCI

Rexben001 committed:
[Feature #163501151] Create method to make user as an Admin

Rexben001 committed:
[Feature #163501151] Make the params user_id

Rexben001 committed:
[Feature #163501151] Create route to make a user an Admin

Rexben001 committed:
[Feature #163501151] Use local database

Rexben001 committed:
[Feature #163501151] Add comments

Rexben001 committed:
[Feature #163501151] Change from post to patch

Rexben001 committed:
feat(register candidate): implement register a candidate
- secure authenticate route
- create table candidates
- Write tests
[Finishes #163501151]
